src/SwedbankPay.Sdk/Amount.cs

```diff
@@ -1,50 +1,113 @@
-
-
-        public long Value { get; }
-
-
-        public static Amount FromDecimal(decimal amount)
-        {
-            return new Amount(amount * 100);
-        }
-
-
-        public static Amount FromInt(long amount)
-        {
-            return new Amount(amount * 100);
-        }
-
-
-        public static decimal ToDecimal(Amount amount)
-        {
-            return (decimal)amount.Value / 100;
-        }
-
-
-        public override string ToString()
-        {
-            return Value.ToString();
+    public class Amount : IEquatable<Amount>, IComparable<Amount>, IComparable
+    {
+        private readonly decimal amount;
+
+        public Amount(decimal amount)
+        {
+            this.amount = amount;
+
+            // Use "Banker's Rounding" by default.
+            const MidpointRounding roundingMode = MidpointRounding.ToEven;
+            var roundedAmount = Math.Round(amount, 2, roundingMode);
+            InLowestMonetaryUnit = (int)(roundedAmount * 100);
+        }
+
+        public int InLowestMonetaryUnit { get; }
+
+        public static implicit operator decimal(Amount amount)
+        {
+            return amount.amount;
+        }
+
+        public static implicit operator Amount(decimal amount)
+        {
+            return new Amount(amount);
+        }
+
+        public static Amount operator +(Amount a, Amount b)
+        {
+            return new Amount(a.amount + b.amount);
+        }
+
+        public static Amount operator -(Amount a, Amount b)
+        {
+            return new Amount(a.amount - b.amount);
+        }
+
+        public static Amount FromLowestMonetaryUnit(int amountInLowestMonetaryUnit)
+        {
+            var amount = (decimal)amountInLowestMonetaryUnit / 100;
+            return new Amount(amount);
+        }
+
+        public int CompareTo(object obj)
+        {
+            if (ReferenceEquals(null, obj))
+                return 1;
+
+            if (ReferenceEquals(this, obj))
+                return 0;
+
+            if (!(obj is Amount))
+                throw new ArgumentException($"Object must be of type {nameof(Amount)}");
+
+            return CompareTo((Amount)obj);
+        }
+
+        public int CompareTo(Amount other)
+        {
+            if (ReferenceEquals(this, other))
+                return 0;
+
+            if (ReferenceEquals(null, other))
+                return 1;
+
+            return amount.CompareTo(other.amount);
+        }
+
+        public bool Equals(Amount other)
+        {
+            if (ReferenceEquals(null, other))
+                return false;
+
+            if (ReferenceEquals(this, other))
+                return true;
+
+            return amount == other.amount;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj))
+                return false;
+
+            if (ReferenceEquals(this, obj))
+                return true;
+
+            if (obj.GetType() != GetType())

```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)